pkg/util/tar_utils.go

```diff
@@ -49,13 +47,16 @@ func unpackTar(tr *tar.Reader, path string) error {
 
 			// Remove the whited-out path.
 			newName := strings.Replace(rmPath, ".wh.", "", 1)
-			if err = os.RemoveAll(newName); err != nil {
+			if err = walkAndRemove(newName); err != nil {
 				logrus.Error(err)
 			}
 			continue
 		}
-
 		target := filepath.Join(path, header.Name)
+		// Make sure the target isn't part of the whitelist
+		if checkWhitelist(target, whitelist) {
```

Comment:
A debug log here will come in handy later.

Reply:
I added it in the checkWhitelist function